include/sys/zfs_rlock.h

```diff
@@ -34,6 +34,24 @@ extern "C" {
 
 #include <sys/zfs_znode.h>
 
+#else
+
+#include <sys/avl.h>
+#include <sys/zfs_context.h>
+#include <sys/refcount.h>
+
+/* User mode znode emulation for ztest */
+typedef struct znode {
+	list_node_t z_lnode;	/* needed by ztest */
+	refcount_t z_refcnt;	/* needed by ztest */
+	uint64_t z_object;	/* needed by ztest */
+	kmutex_t z_range_lock;
+	avl_tree_t z_range_avl;
+	boolean_t z_is_zvol;
+} znode_t;
```

Comment:
I'd consider moving this user space `znode_t` definition in to `zfs_znode.h`.  That should allow you to entirely from the `#ifdef _KERNEL` wrapper from `zfs_rlock.h` which would be nice.  And it would have the benefit of having both versions of `znode_t` be declared in on place.